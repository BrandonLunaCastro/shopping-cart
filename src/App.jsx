import Header from "./components/Header/Header";
import Router from "./router/router";

function App() {
  return (
    <>
      <Header />
    </>
  );
}

export default App;
