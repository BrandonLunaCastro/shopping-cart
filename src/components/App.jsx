import Root from "../router/Root";
import { useState } from "react";

function App() {
  const [articles, setArticles] = useState(0);
  const [added, setAdded] = useState([]);
  const [data, setData] = useState([]);

  return (
    <Root
      articles={articles}
      setArticles={setArticles}
      setAdded={setAdded}
      added={added}
      setData={setData}
      data={data}
    ></Root>
  );
}

export default App;
