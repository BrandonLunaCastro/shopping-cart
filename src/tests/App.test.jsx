import {  beforeEach, describe, expect, it, vi } from "vitest";
import App from "../components/App";
import {  act, render, screen } from "@testing-library/react";
import ShoppingProvider from "../context/ShoppingCartContext";
import userEvent from "@testing-library/user-event";


beforeEach(() => {
  act(() => {
    render(
      <ShoppingProvider>
        <App /> 
      </ShoppingProvider>
    );
  })
});

describe("Should be render App", () => {
  it("should be render first a Home component", () => {
    const title = screen.getByText(/You are in Home/i)
    expect(title).toBeInTheDocument();
  })

  it("should be render Store component when link is clicked", async () => {
    const user = userEvent.setup();
    const storeLink = screen.getByText(/Store/i);

    await user.click(storeLink)
    expect(screen.getByRole("heading", {name: "Main articles" }))
  })

  it("should  be open the cart", async () => {
    const user = userEvent.setup();
    const cartBtn = await screen.findByAltText("cart-icon");
    await user.click(cartBtn);                     
    expect(screen.getByText(/the actual cart is empty/i)).toMatchSnapshot();
  })

  it("should be add article", async () => {
    const user = userEvent.setup();
    const button = await screen.findByTestId("btn-2")
    await user.click(button);
    
    user.click(screen.getByAltText("cart-icon"));
    expect(screen.getAllByText(/mens casual/i))
  })

  it("should be add article on view of Snapshot", async () => {
    const user = userEvent.setup();
    const button = await screen.findByTestId("btn-2")
    await user.click(button);
    
    user.click(screen.getByAltText("cart-icon"));
    expect(screen.getAllByText(/mens casual/i)).toMatchSnapshot()
  })

  it("should be length cart items 1", async () => {
    const user = userEvent.setup();
    const button = await screen.findByTestId("btn-3");
    act(() => {
      user.click(button);
      user.click(screen.getByAltText("cart-icon"));
    })
    // expect(screen.getAllByTestId("article-cart")).toHaveLength(1);
    const articles =  screen.queryAllByTestId("article-cart"); 
    expect(articles.length).toBe(1);
    // expect(screen.getByText(/mens cotton/i)).toBeInTheDocument();
    // expect(screen.getByText("Articles Added")).toBeInTheDocument();
  });


});
