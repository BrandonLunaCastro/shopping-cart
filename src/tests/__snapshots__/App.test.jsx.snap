// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should be render App > should  be open the cart 1`] = `
<p>
  The actual Cart is empty
</p>
`;

exports[`Should be render App > should be add article on view of Snapshot 1`] = `
[
  <p>
    Mens Casual Premium Slim Fit T-Shirts 
  </p>,
  <p>
    Mens Casual Slim Fit
  </p>,
]
`;
